CONTRIBUTING.md

```diff
@@ -1,44 +1,199 @@
+
+To submit new question in GitHub Issues, it must include these details:
+
+- **Title**
+    - Write prefix `[Question]`
+    - Use a clear and descriptive title
+- **Body** - describe your question with as many details as possible.
+
+
+
+### Your First Code Contribution
+
+Read our [C++ Style Guide](#c++-style-guide) and start with beginner-friendly issues with label [`[good-first-issue]`](https://github.com/hyperledger/iroha/issues?q=is:open+is:issue+label:good-first-issue ). 
+
+
+
+### Pull Requests
+
+- Fill in [the required template](.github/PULL_REQUEST_TEMPLATE.md)
+
+- **Write tests** for new code.
+
+- Every pull request should be reviewed and **get at least two approvals**.
+
+- Do not include issue numbers in the PR title or commit messages.
+
+- Include issue numbers in Pull Request body only.
+
+- When finished work, **rebase onto base branch** with 
+    ```bash
+    $ git fetch
+    $ git rebase -i <base-branch>
+    ```
+
+    [Step-by-step guide](https://soramitsu.atlassian.net/wiki/spaces/IS/pages/11173889/Rebase+and+merge+guide).
+
+- Follow the [C++ Style Guide](#C++-style-guide).
+
+- Follow the [Git Style Guide](#git-commit-messages) .
+
+- **Document new code** based on the [Documentation Styleguide](#documentation-styleguide)
+
+- End all files with a newline.
+
+
+
+
+## Styleguides
+
+### Git Style Guide
+
+- **Use present tense** ("Add feautre", not "Added feature").
+- **Use imperative mood** ("Deploy docker to..." not "Deploys docker to...").
+- Write meaningful commit message.
+- **Sign every commit** with [DCO](https://github.com/apps/dco): `Signed-off-by: $NAME <$EMAIL>`. 
+    You can do it automatically using `git commit -s`.
+- Do not include PR or Issue number in commit message. 
+- Limit the first line of commit message to 50 characters or less.
+- When only changing documentation, include `[ci skip]` in the commit description.
+- We use mixed approach of [Github Flow](https://guides.github.com/introduction/flow/) and [Git Flow](http://nvie.com/posts/a-successful-git-branching-model/). More at [Iroha Working Agreement](https://github.com/hyperledger/iroha/wiki/Iroha-working-agreement#2-version-control-system).
+
+
+
+
+### C++ Style Guide
+
+- Use [clang-format](http://clang.llvm.org/docs/ClangFormat.html) for code formatting. 
+- Follow [CppCoreGuidelines](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines) and [Cpp Best Practices](https://lefticus.gitbooks.io/cpp-best-practices).
+- Avoid platform-dependent code.
+- Use [C++14](https://en.wikipedia.org/wiki/C%2B%2B14).
+- Use [camelCase](https://en.wikipedia.org/wiki/Camel_case).
+
+
+
+### Documentation Styleguide
+
+- Use [Doxygen](http://www.stack.nl/~dimitri/doxygen/manual/docblocks.html).
+- Document all public API: methods, functions, members, templates, classes...
+
+
+
+## Additional Notes
+
+### Issue and Pull Request Labels
```

Comment:
Maybe it's better separate it? E.g `question` and `good-first-issue` are only applicable to issues

Reply:
As I see all of the labels applicable to question, so it's possible to make a `PR labels` section and `other labels` section

Reply:
Also it's better place it in other file, usually labels is set by maintainers